pkg/audit/manager.go

```diff
@@ -458,14 +459,27 @@ func (am *Manager) auditFromCache(ctx context.Context) ([]Result, []error) {
 	if err != nil {
 		return nil, []error{fmt.Errorf("unable to list objects from audit cache: %w", err)}
 	}
+	nsMap, err := nsMapFromObjs(objs)
+	if err != nil {
+		return nil, []error{fmt.Errorf("unable to build namespaces from cache: %w", err)}
+	}
 
 	var results []Result
 
 	var errs []error
 	for i := range objs {
 		// Prevent referencing loop variables directly.
 		obj := objs[i]
-		resp, err := am.opa.Review(ctx, obj)
+		ns, exists := nsMap[obj.GetNamespace()]
+		if !exists {
+			ns = nil
+		}
+
+		au := &target.AugmentedUnstructured{
+			Object:    obj,
+			Namespace: ns,
+		}
+		resp, err := am.opa.Review(ctx, au)
```

Comment:
Feel free to ignore, but I am curious how/ if we'd wanna test this.

Putting on a TDD hat 🧢 , it'd be 💯 to add some form of test to A) catch any regressions on this issue and B) programmatically prove the issue exists and this patch solves it. 

I was poking around to find a good place to unit test this but I am still getting familiar w the code here 🤷🏼 ; so I can't offer more specific guidance 😞

Reply:
Easiest would be to add an e2e test for this use case, but that wouldn't be trivial, as it would involve installing G8r with a second set of flags (doable, but more work than just adding an e2e test case).

Another option would be to mock the various k8s clients and use dependency injection to create unit tests. This is preferable from a test runtime perspective, but would involve significant mocking.

A third option would be to use a local instance of an API server, similar to how [controller tests work](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/controller/constrainttemplate/constrainttemplate_controller_test.go)

TBH the paucity of audit tests is an ongoing issue (currently e2e tested), and I'd prefer to not block a bugfix on what would probably be significant dev. time.